mypy/checker.py

```diff
@@ -643,9 +643,17 @@ def _visit_overloaded_func_def(self, defn: OverloadedFuncDef) -> None:
         if defn.impl:
             defn.impl.accept(self)
         if defn.info:
-            found_base_method = self.check_method_override(defn)
-            if defn.is_explicit_override and found_base_method is False:
+            found_method_base_classes = self.check_method_override(defn)
+            if defn.is_explicit_override and not found_method_base_classes:
```

Comment:
`not found_method_base_class` also is `True` if `found_method_base_classes` is `None`, which is not what we want, I think. Because `None` indicates that another pass has to be made.

```suggestion
            if defn.is_explicit_override and found_method_base_classes is not None and not found_method_base_classes:
```

Reply:
I think you can do this more elegantly by changing the order of the if-branches.